test/integration/windows/windows_test.go

```diff
@@ -74,6 +85,76 @@ var _ = Describe("Windows Integration Test", func() {
 	JustAfterEach(func() {
 		// Will clean up all the resources used by a test
 		frameWork.NSManager.DeleteAndWaitTillNamespaceDeleted(ctx, namespace)
+		configMapWrapper.DeleteConfigMap(frameWork.K8sClient, ctx, configmap)
+	})
+
+	Describe("windows IPAM tests", func() {
+		var deployment *appsV1.Deployment
+
+		BeforeEach(func() {
+
+			deployment = manifest.NewWindowsDeploymentBuilder().
```

Comment:
We already have these sort of tests, can't we reuse those instead of a new one?

Reply:
We could test with a single Pod instead of deployment as we are just testing the feature flag.

- flag=false - flag=true, Pod create successfully.
- flag=false, Pod fail to create.
- No ConfigMap, Pod fail to create
- ConfigMap with No Data field, Pod fail to crate
- ConfigMap with Data Field but wrong value, Pod fail to create.

We also need to verify the Pod doesn't have the ResourceLimit injected when the flag is not true.

Reply:
Added the new test cases, please have a look.